core/src/main/scala/kafka/tools/JmxTool.scala

```diff
@@ -80,8 +83,32 @@ object JmxTool extends Logging {
     val attributesWhitelist = if(attributesWhitelistExists) Some(options.valueOf(attributesOpt).split(",")) else None
     val dateFormatExists = options.has(dateFormatOpt)
     val dateFormat = if(dateFormatExists) Some(new SimpleDateFormat(options.valueOf(dateFormatOpt))) else None
-    val jmxc = JMXConnectorFactory.connect(url, null)
-    val mbsc = jmxc.getMBeanServerConnection()
+
+    var jmxc : JMXConnector = null
+    var mbsc : MBeanServerConnection = null
+    var retries = 0
+    val maxNumRetries = 20
+    var connected = false
+    while (retries < maxNumRetries && !connected) {
```

Comment:
Maybe we should add a comment in the class javadoc about the fact that this tool only works reliably if invoked after the JMX server is fully initialised.